src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -75,9 +75,17 @@ private unsafe static int CompareOrdinalIgnoreCaseHelper(String strA, String str
 
         private unsafe static bool EqualsHelper(String strA, String strB)
         {
-            Contract.Requires(strA != null);
-            Contract.Requires(strB != null);
-            Contract.Requires(strA.Length == strB.Length);
+            Debug.Assert((strA == null || strB == null) || strA.Length == strB.Length);
+
+            if ((object)strA == strB)
```

Comment:
It is code-size tradeoff. You are making the code for all places that compare strings bigger. How do you know that it is a good tradeoff for programs out there?